docs/internal/release-procedure.md

```diff
@@ -174,3 +174,156 @@ git push origin version-<YOUR_VERSION>
+    - `MACOS_FORCE_NOTARIZE`: "true"
+
+    ```sh
+    export AC_PASSWORD="***"
+    export AC_USERNAME="***"
+    export AC_TEAM_ID="7KT7ZWMCJT"
+    export CSC_KEY_PASSWORD="***"
+    export CSC_LINK="`pwd`/Pro-IDE-Certificates.p12"
+    export IS_RELEASE="true"
+    export CAN_SIGN="true"
+    export MACOS_FORCE_NOTARIZE="true"
+    export CI="true"
+    ```
+
+- Build:
+
+  ```sh
+  ./scripts/package.sh
+  ```
+
+  > Make sure to change directory, and execute the script from the root of the `arduino-ide` repository.
+
+- Verify:
+
+  Since you cannot drag and drop via TeamViewer, you will install the app from a command line. This example puts the IDE2 into the Desktop. Do **NOT** try to bypass the installation with a double click and open from the UI. The `node_modules` folder of the source code is implicitly in the `$PATH`, and you want to verify if the app is fully functional without the `node_modules` folder.
+
+  ```sh
+  VERSION="<TODO: release version>"
+  hdiutil attach ./electron-app/dist/arduino-ide_${VERSION}_macOS_arm64.dmg \
+  && cp -R /Volumes/Arduino\ IDE\ ${VERSION}-arm64/Arduino\ IDE.app ~/Desktop \
+  && hdiutil unmount /Volumes/Arduino\ IDE\ ${VERSION}-arm64 \
+  && codesign -dv --verbose=4 ~/Desktop/Arduino\ IDE.app \
+  && ~/Desktop/Arduino\ IDE.app/Contents/MacOS/Arduino\ IDE
+  ```
+
+- Cleanup:
+
+  - You **MUST** close the shell after the build.
+  - You **MUST** delete the `.p12` file and empty the trash afterward.
+
+- Artifacts:
+  - You have to upload the following artifacts from `./arduino-ide/electron-app/dist` to S3, but first, you must create the final channel file from the `latest-mac.yaml`:
+    - `arduino-ide_${VERSION}_macOS_arm64.dmg`,
+    - `arduino-ide_${VERSION}_macOS_arm64.zip`, and
+    - `stable-mac.yml`
+  - To create the final channel file, do the followings:
+    - Copy the `stable-mac.yml` file from the Mac Mini to a folder and rename it to `stable-mac-ARM64.yml`.
+    - Download the `stable-mac.yaml` produced by GitHub Actions from the latest release, rename it to `stable-mac-X64.yml`, and put it in the same folder where you put the file from the Mac Mini.
+    - Run the channel file merger:
+      ```sh
+      node ./scripts/merge-channel-files.js ./path/to/folder/with/channel/files --no-cleanup
+      ```
+      > Although, the `--no-cleanup` flag is not mandatory to merge the channel files, it instructs the merger to keep the original channel files.
+    - You have the merged channel file that you need to upload to S3.
+
+### FAQ
+
+- Q: I see no `stable` channel files, only `latest`.
+- A: You forgot to set the `CI=true` environment variable.
+
+---
+
+- Q: How to connect to the EC2 instance?
+- A: DevOps will give you a temporary link to the private key. Create a file `username_ip.pem` in your cwd, copy the private key into the file, open a shell, and execute `ssh -i "username_ip.pem" username@ip`. DevOps will tell you the `username` and the `ip`. Do not forget the VPN.
```

Comment:
The `chmod 0400 username_ip.pem` is missing from the steps. Otherwise there is an error:


```
Load key "username_ip.pem": bad permissions
username@ip: Permission denied (publickey).
```